src/string.ts

```diff
@@ -16,3 +16,80 @@ export function parseBoolean(candidate: string | number) {
     return false;
   }
 }
+
+/**
+ * Splits a string into an array of words, respecting quoted phrases.
+ * Handles comma-separated values where phrases in double quotes are treated as a single unit,
+ * even if they contain commas.
+ *
+ * This is basically CSV parsing, if we find this to be cumbersome to maintain,
+ * we can consider using a dedicated CSV parsing library.
+ *
+ * Example: '"apple, banana", cherry' => ['apple, banana', 'cherry']
+ *
+ * @param {string} candidate - The input string to split
+ * @return {string[]} - Array of split words/phrases
+ */
+export function splitWords(candidate: string): string[] {
+  // Handle empty input case
+  if (!candidate) return [''];
+
+  const result = [];
+  let currentWord = '';
+  let inQuotes = false; // Tracks whether we're currently inside a quoted section
+
+  // Process the string character by character:
+  // 1. Toggle quote state when we encounter double quotes (") - marking quoted sections
+  //    Quotes themselves are not included in the final words
+  // 2. When we hit a comma outside quotes, it's treated as a word separator:
+  //    - Current word is trimmed and added to results array
+  //    - Current word buffer is reset for the next word
+  // 3. Inside quoted sections, commas are treated as regular characters (part of the word)
+  // 4. All other characters are added to the current word being built
+  for (let i = 0; i < candidate.length; i++) {
+    const char = candidate[i];
+
+    if (char === '"') {
+      inQuotes = !inQuotes;
+      continue; // Skip adding the quote character to the result
+    }
+
+    if (char === ',' && !inQuotes) {
+      result.push(currentWord.trim()); // Add the completed word to results
+      currentWord = ''; // Reset for the next word
+      continue;
+    }
+
+    currentWord += char;
+  }
+
+  // Check for mismatched quotes
+  if (inQuotes) {
+    throw new Error('Mismatched quotes in input string');
+  }
```

Comment:
Do we need to handle this? A query with a single quote should be valid right?

Reply:
It made sense to do this, although it makes the `"` an invalid search candidate, I think it still works. 

Since there's not visual feedback on the frontend about how the values were sent, if we allow mismatched quote, we will have to break the sentence through some fallback method which will introduce ambiguity.

Reply:
@pranavrajs let me know what you think here, we can proceed with merging this